Userland/Libraries/LibPDF/Fonts/PS1FontProgram.cpp

```diff
@@ -0,0 +1,555 @@
+    return parse_encrypted_portion(decrypted);
+}
+
+Gfx::Path PS1FontProgram::build_char(u32 code_point, Gfx::FloatPoint const& point, float width)
+{
+    if (!m_glpyh_map.contains(code_point))
+        return {};
+
+    auto glyph = m_glpyh_map.get(code_point).value();
+    auto scale = width / m_font_matrix.map(glyph.width).x();
+    auto transform = m_font_matrix;
+
+    // FIXME: Why do we need to flip the glyph here?
+    transform.scale(scale, -scale);
+    transform.set_translation(point);
+
+    return glyph.path.copy_transformed(transform);
+}
+
+PDFErrorOr<PS1FontProgram::Glyph> PS1FontProgram::parse_glyph(ReadonlyBytes const& data, GlyphParserState& state)
+{
+    auto push = [&](float value) -> PDFErrorOr<void> {
+        if (state.sp >= state.stack.size())
+            return error("Operand stack overflow");
+        state.stack[state.sp++] = value;
+        return {};
+    };
+
+    auto pop = [&]() -> float {
+        return state.sp ? state.stack[--state.sp] : 0.0f;
+    };
+
+    auto& path = state.glyph.path;
+
+    // Parse the stream of parameters and commands that make up a glyph outline.
+    for (size_t i = 0; i < data.size(); ++i) {
+        auto require = [&](unsigned num) -> PDFErrorOr<void> {
+            if (i + num >= data.size())
+                return error("Malformed glyph outline definition");
+            return {};
+        };
+
+        int v = data[i];
+        if (v == 255) {
+            TRY(require(4));
+            int a = data[++i];
+            int b = data[++i];
+            int c = data[++i];
+            int d = data[++i];
+            TRY(push((a << 24) + (b << 16) + (c << 8) + d));
+        } else if (v >= 251) {
+            TRY(require(1));
+            auto w = data[++i];
+            TRY(push(-((v - 251) * 256) - w - 108));
+        } else if (v >= 247) {
+            TRY(require(1));
+            auto w = data[++i];
+            TRY(push(((v - 247) * 256) + w + 108));
+        } else if (v >= 32) {
+            TRY(push(v - 139));
+        } else {
+            // Not a parameter but a command byte.
+            switch (v) {
+            case 1: // y dy hstem
+            case 3: // x dx vstem
+                state.sp = 0;
+                break;
+
+            case 4: { // dy vmoveto
+                auto dy = pop();
+
+                state.point.translate_by(0.0f, dy);
+
+                if (state.flex_feature) {
+                    state.flex_sequence[state.flex_index++] = state.point.x();
+                    state.flex_sequence[state.flex_index++] = state.point.y();
+                } else {
+                    path.move_to(state.point);
+                }
+                state.sp = 0;
+            } break;
+
+            case 5: { // dx dy rlineto
+                auto dy = pop();
+                auto dx = pop();
+
+                state.point.translate_by(dx, dy);
+                path.line_to(state.point);
+                state.sp = 0;
+            } break;
```

Comment:
`break` typically goes inside of the braces in Serenity code.

Reply:
oops, forgot that. Fixed it just now.